src/main/java/com/mongodb/jdbc/MongoSQLStatement.java

```diff
@@ -31,10 +34,30 @@ public ResultSet executeQuery(String sql) throws SQLException {
                 iterable = iterable.batchSize(fetchSize);
             }
 
-            resultSet = new MongoSQLResultSet(this, iterable.cursor());
+            MongoIterable<MongoJsonSchemaResult> schemaIterable =
+                    currentDB
+                            .withCodecRegistry(MongoDriver.registry)
+                            .aggregate(
+                                    Collections.singletonList(schemaStage),
```

Comment:
Have you run this against an adl yet? I don't think this should be an `aggregate`. the `sqlGetResultSchema` command is not an aggregation stage. You should use `runCommand` instead (specifically this version: `<TResult> TResult runCommand(Bson var1, Class<TResult> var2)`).

Reply:
I will update to using runCommand().  Thanks for the suggestion of running against ADL, I was able to run the command and it reached ADL and returned an error.  This made it clearer how the errors would be propagated to the user.